debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/FindInstances.java

```diff
@@ -0,0 +1,159 @@
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ */
+package com.ibm.j9ddr.vm29.tools.ddrinteractive.commands;
+
+import java.io.PrintStream;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.tools.ddrinteractive.CommandUtils;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.tools.ddrinteractive.Table;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ObjectPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ClassHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+
+import com.ibm.j9ddr.vm29.j9.DataType;
+import com.ibm.j9ddr.vm29.j9.walkers.ClassSegmentIterator;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+
+import com.ibm.j9ddr.vm29.j9.gc.GCHeapRegionDescriptor;
+import com.ibm.j9ddr.vm29.j9.gc.GCHeapRegionIterator;
+import com.ibm.j9ddr.vm29.j9.gc.GCObjectHeapIterator;
+
+public class FindInstances extends Command
+{
+	private String className;
+	private J9ClassPointer classPointer;
+	private Table data;
+	private long corruptCount;
+	private long objectsFound;
+
+	public FindInstances() {
+		addCommand("findinstances", "<classname>", "find all instances of the specific class and its subclasses");
+	}
+
+	private static void printUsage (PrintStream out) {
+		out.println("findinstances <name> - find all instances of the specific class and its subclasses");
+	}
+
+	private boolean parseArgs(PrintStream out, String[] args) throws DDRInteractiveCommandException {
+		if (args.length != 1) {
+			out.println("Exactly one argument expected");
+			return false;
+		}
+		String firstArg = args[0];
+		if (firstArg.equals("help")) {
+			printUsage(out);
+			return false;
+		}
+		className = firstArg;
+		return true;
+	}
+
+	@Override
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException {
+		if (args != null) {
+			if (!parseArgs(out, args)) {
```

Comment:
If `args` is null, I don't think we want to reuse the same arguments from the last use. Suggest moving the null check into `parseArgs()`:
```java
		if ((args == null) || (args.length != 1)) {
			out.println("Exactly one argument expected");
			return false;
		}
```